File: peek_plugin_chat/_private/tuples/ChatReadActionTuple.py
from datetime import datetime

from peek_plugin_chat._private.PluginNames import chatTuplePrefix
from vortex.Tuple import addTupleType, TupleField
from vortex.TupleAction import TupleActionABC


@addTupleType
class ChatReadActionTuple(TupleActionABC):
    __tupleType__ = chatTuplePrefix + "ChatReadActionTuple"

    chatUserId = TupleField(typingType=int)
    readDateTime = TupleField(typingType=datetime)

File: peek_plugin_chat/plugin-module/_private/tuples/ChatReadActionTuple.ts
import {addTupleType, TupleActionABC} from "@synerty/vortexjs";
import {chatTuplePrefix} from "../PluginNames";

@addTupleType
export class ChatReadActionTuple extends TupleActionABC {
    static readonly tupleName = chatTuplePrefix + "ChatReadActionTuple";

    chatUserId: number;
    readDateTime: Date;

    constructor() {
        super(ChatReadActionTuple.tupleName)
    }
}